consul/autopilot.go

```diff
@@ -113,13 +82,18 @@ func (s *Server) pruneDeadServers() error {
 		}
 	}
 
+	// Nothing to remove, return early
+	if len(failed) == 0 {
+		return nil
+	}
+
 	peers, err := s.numPeers()
 	if err != nil {
 		return err
 	}
 
 	// Only do removals if a minority of servers will be affected
-	if len(failed) <= peers/2 {
+	if len(failed) < peers/2 || (len(failed) == 1 && peers >= 3) {
```

Comment:
I've kind of gone back and forth, but it seems like we should be more conservative to start. This periodic check is a good cleanup for existing dead servers or if we weren't able to delete it when the new server was added. There's no downside for leaving it in there while we wait for one to be added, though.